proto/coroutine/v1/type.proto

```diff
@@ -47,17 +48,17 @@ message Type {
   // Variadic is true for function types with a variadic argument.
   bool variadic = 12;
 
-  // CustomSerializer is true if the type has a custom serializer registered.
-  bool custom_serializer = 13;
+  // Custom is true if the type is a custom opaque type.
+  bool custom = 13;
 }
 
 message Field {
-  // Name is the name of the field.
-  string name = 1;
+  // Name is the string ID of the name of the field.
+  int32 name = 1;
```

Comment:
Is there a reason for using signed integers here?

Reply:
No reason. I think originally I was going to use negative values as sentinels (e.g. -1 = nil), but we're using 0 for that so no reason to have signed types.